### PR TITLE
ConcreteSeries for Series based on Expressions

### PR DESCRIPTION
**Pull request by adrianmroz**

### CODE REVIEW

File: src/common/models/series/concrete-series.ts [hunk 1]
```diff
@@ -21,41 +21,39 @@ import { TimeShiftEnv } from "../time-shift/time-shift-env";
...
+        return $("main").filter(filter);
       }
-    }
+      return null;
+    });
+  }
 
-    return `${derivationTitle(derivation)}${this.measure.title}`;
+  protected applyPeriod(derivation: MeasureDerivation, timeShiftEnv: TimeShiftEnv): Expression {
+    if (derivation === MeasureDerivation.CURRENT) return this.measure.expression;
```
Comment (adrianmroz) [inline src/common/models/series/concrete-series.ts]: WRONG: if there's no time shift, return expression, when is - apply filter from TimeShiftEnv depending on derivation.

File: src/common/models/series/concrete-series.ts [hunk 2]
```diff
@@ -71,28 +69,33 @@ export class ConcreteSeries {
...
+    const value = this.selectValue(datum, period);
+    const formatter = seriesFormatter(this.series.format, this.measure);
+    return formatter(value);
   }
 
-  public plywoodExpression(nestingLevel: number, derivation: MeasureDerivation, timeShiftEnv: TimeShiftEnv): ApplyExpression {
-    const expression = this.applyPeriod(derivation, timeShiftEnv);
-    const name = this.plywoodKey(derivation);
-    return new ApplyExpression({ name, expression });
+  private derivationTitle(derivation: MeasureDerivation): string {
```
Comment (adrianmroz) [inline src/common/models/series/concrete-series.ts]: import from measure.ts?

File: src/client/components/dimension-tile/dimension-tile.tsx [hunk 3]
```diff
@@ -176,8 +175,7 @@ export class DimensionTile extends React.Component<DimensionTileProps, Dimension
     }
 
     if (sortOn.reference instanceof Measure) {
-      const measure = sortOn.reference;
-      const series = new ConcreteSeries(MeasureSeries.fromMeasure(measure), measure);
+      const series = fromMeasure(sortOn.reference);
       // TODO: FIX that
       query = query.performAction(series.plywoodExpression(0, MeasureDerivation.CURRENT, null));
```
Comment (adrianmroz) [inline src/client/components/dimension-tile/dimension-tile.tsx]: encapsulate this creation of series and conversion to plywood expression

File: src/common/models/series/measure-concrete-series.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  return new MeasureConcreteSeries(MeasureSeries.fromMeasure(measure), measure);
+}
+
+export class MeasureConcreteSeries extends ConcreteSeries<MeasureSeries> {
+
+  constructor(series: MeasureSeries, measure: Measure) {
+    super(series, measure);
+  }
+
+  public plywoodExpression(nestingLevel: number, derivationFilter?: DerivationFilter): ApplyExpression {
```
Comment (mkuthan) [inline src/common/models/series/measure-concrete-series.ts]: nestingLevel parameter is not used, it looks that abstraction (ConcreteSeries) is somehow not cohesive

File: src/common/models/series/expression-concrete-series.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+  key(derivation?: SeriesDerivation): string {
+    return `${super.key(derivation)}-${this.series.operation}`;
+  }
+
+  title(derivation?: SeriesDerivation): string {
+    return `${super.title(derivation)} ${this.operationTitle()}`;
+  }
+
+  private operationTitle(): string {
```
Comment (mkuthan) [inline src/common/models/series/expression-concrete-series.ts]: it could be a part of ExpressionSeries

**Pull request closed without merging**